keps/sig-storage/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,681 @@
+
+<!--
+This section is incredibly important for producing high quality user-focused
+documentation such as release notes or a development roadmap.  It should be
+possible to collect this information before implementation begins in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself.  KEP editors, SIG Docs, and SIG PM
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+Currently, the PVCs created automatically by the StatefulSet are not deleted when 
+the StatefulSet is deleted. As can be seen by the discussion in the issue 
+[55045](https://github.com/kubernetes/kubernetes/issues/55045) there are several use
+cases where the PVCs which are automatically created are deleted as well. There is
+merit to the suggestion that the PVCs should not be deleted since they have a different
+lifecycle than the pods, hence this proposal intends to keep this as an opt in 
+for users.
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+Provide a feature to auto delete the PVCs created by StatefulSet.
+
+<!--
+List the specific goals of the KEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+This proposal does not plan to address how the underlying PVs are treated on PVC deletion. 
+That functionality will continue to be governed by the ReclaimPolicy of the storage class. 
+
+<!--
+What is out of scope for this KEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+### Background
+
+Controller `garbagecollector` is responsible for ensuring that when a statefulset 
+set is deleted the corresponding pods spawned from the StatefulSet is deleted. 
+The `garbagecollector` uses `OwnerReference` added to the `Pod` by statefulset controller
+to delete the Pod. Similar mechanism is leveraged by this proposal to automatically 
+delete the PVCs created by the StatefulSet controller.
+
+### Changes required
+
+The following changes are required:
+
+1. Add `VolumeReclaimPolicy` entry into StatefulSet spec inorder to make this feature an opt-in.
```

Comment:
the word VolumeReclaimPolicy is confusing, it seems to conflict in meaning with reclaimPolicy in PV. Should we make it more clear pvcReclaimPolicy ?

Reply:
I think that's a good point.

Reply:
+1